crates/router/src/connector/adyen/transformers.rs

```diff
@@ -604,6 +613,24 @@ pub struct AliPayData {
     payment_type: PaymentType,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AliPayHkData {
+    #[serde(rename = "type")]
+    payment_type: PaymentType,
+}
```

Comment:
How does the request sent to Adyen look like? I see a `#[serde(tag = "type")]` annotation on the `AdyenPaymentMethod` type, and a `#[serde(rename = "type")]` annotation here. Aren't they kinda redundant?

Reply:
Yes `#[serde(rename = "type")]` is a redundant annotation. But all the payment methods implemented for Adyen uses it. Can I raise a seperate PR to fix it along with other general fix

Reply:
Yeah, it should be fine to take this up in a separate PR.